document-sandbox-samples/image-and-page/src/code.js

```diff
@@ -60,19 +60,21 @@ async function start() {
         createImage: async function(blob, size = {}) {
             const insertionParent = editor.context.insertionParent;
             const bitmapImage = await editor.loadBitmapImage(blob);
-            let { width, height } = size;
-            if (!width || !height) {
-                width = bitmapImage.width;
-                height = bitmapImage.height;
-            }
-            const mediaContainerNode = editor.createImageContainer(bitmapImage, { initialSize: { width, height } });
-            insertionParent.children.append(mediaContainerNode);
+            await editor.queueAsyncEdit(() => {
```

Comment:
An explanatory comment above this would be good. Maybe like:
```
// Edits following an asynchronous wait need to be queued to run at a safe time
```